provisioning-workflows/zOS/NetworkResources/Provision_Network_Resource.xml

```diff
@@ -0,0 +1,252 @@
+        <description>The object id of the reserved Port address to use during deprovisioning</description>
+        <category>REST</category>
+        <string/>
+    </variable>
+    <variable name="APPLID" visibility="public">
+        <label>APPLID</label>
+        <abstract>VTAM APPLID</abstract>
+        <description>VTAM APPLID</description>
+        <category>REST</category>
+        <string/>
+    </variable>
+    <variable name="APPLID_ID" visibility="private">
+        <label>APPLID_ID</label>
+        <abstract>Object ID of the APPLID resource</abstract>
+        <description>The object id of the reserved APPLID resource to use during deprovisioning</description>
+        <category>REST</category>
+        <string/>
+    </variable>
+    <variable name="JOB_NAME" visibility="public">
+        <label>JOB_NAME</label>
+        <abstract>Job name of server that will use allocated port. Specify * if Jobname is not available</abstract>
+        <description>Job name of server that will use allocated port. Specify * if Jobname is not available</description>
+        <category>REST</category>
+        <string valueMustBeChoice="false" multiLine="false"/>
+    </variable>
+    <variable name="USAGE_TYPE" visibility="public">
+        <label>USAGE_TYPE</label>
+        <abstract>Usage type that matches with usage type associated with port pool</abstract>
+        <description>Usage type that matches with usage type associated with port pool. Request will fail if matching usage type is not found</description>
+        <category>REST</category>
+        <string valueMustBeChoice="false" multiLine="false"/>
+    </variable>
+
+    <atCreate name="OBTAIN_DVIPA" scope="instance" required="false" prompt="true"/>
+    <atCreate name="OBTAIN_PORT" scope="instance" required="false" prompt="true"/>
+    <atCreate name="OBTAIN_APPLID" scope="instance" required="false" prompt="true"/>
+    <atCreate name="JOB_NAME" scope="instance" required="false" prompt="true"/>
+    <atCreate name="USAGE_TYPE" scope="instance" required="false" prompt="true"/>
+
+    <step name="allocateNetworkResources">
+        <title>Allocate Network Resources</title>
+        <description>Allocate port and/or DVIPA and/or VTAM APPLID</description>
+        <step name="allocatePort" optional="false">
+            <title>Allocate Port</title>
+            <description>Make a REST call to allocate port</description>
+            <condition>
+                <expression>"1" == "1"</expression>
+                <description>Check that previous step has completed</description>
+                <targetStateSet>
+                    <description>Set state to Skipped if TCP/IP port resource
+                         is not requested
+                    </description>
+                    <extendStateExpression>
+                        <description>Skip if OBTAIN_PORT does not equal YES</description>
+                        <expression>${instance_OBTAIN_PORT} != "YES"</expression>
```

Comment:
This is a general comment that applies to all these variables used in these sorts of expressions.
Is there any reason we are doing a string match on "YES" instead of having these variables as booleans? They seem to only be used for expression checks. 
If there is a reason for using strings, I have an additional suggestion, but I think these should probably be booleans.

Reply:
It does not matter if there is string compare or boolean. "YES"/"NO" reads better on choice to obtain resource vs TRUE/FALSE

Reply:
Functionally,I agree, it doesn't matter in this application.
If the purpose of this workflow is to be an example I don't think using YES/NO and doing a string compare for variables that describe themselves as booleans is the best example to make.

Reply:
It's more readable for choice than True/False boolean. "Obtain_Port" value "No" vs "False" is more intuitive.
If value is specified via property file, we can define variable as boolean. But in this case, string makes more sense. So, lets go with that.